### ISSUE TITLE
New functionality for nilearn.image.resample_img

### ISSUE THREAD

**Issue by anbai106** [nilearn/nilearn#1801]
Hello,

I am struggling using 

> nilearn.image.resample_img

to crop my MRI within a bounding box. For example, I give a range of indexes for sagittal, coronal and axial axes. I know I can use the argument of target_size in resample_img to have the same dimension. But I can not choose the range of indexes of slices in each axis. 

Alternaltively, I tried Nibabel to crop the MRI manually, however, the problem for Nibabel is that I have to change the header to adjust the new cropped MRI, which is out of scope my knowledge.

I would like to ask you if you can add such functionality in your library to crop MRI with a boding box with adaptation of the change of the header, thus the cropped MRI will be in the same space (coordinate) as the original MRI.

Thanks in advance

Hao

**Comment by anbai106** [comment]
@banilo Thanks for your quick reply, this function is great, but I think we define this parameter **rtol** (for example, troll=0.01) to control the edge of the cropping image, right? Actually, right now, I have all my individual MRI registered into MNI space with the same dimensions. I just want reduce the dimension of all the individual MRI with a common bounding box. I think the function you offered works for each individual MRI, but I am not sure if it will result in the same image size for all individual MRI…

Hao

**Comment by bthirion** [comment]
Sure, you do the corpping "manually" (as I indicated previously) on one image, and then you use resample_to_img to crop all the others to the same FOV.
Does that answer you question ?

**Comment by anbai106** [comment]
@bthirion Thanks very much, that is exactly what I want !!!

**Comment by anbai106** [comment]
@GaelVaroquaux  Good idea, this could be a good utility for the community, especially for those who do not really know the MRI header like me...

**Comment by kchawla-pi** [comment]
Would a wrapper function of some kind, doing all of this with a single command be feasible & useful?

**Comment by GaelVaroquaux** [comment]
> Would a wrapper function of some kind, doing all of this with a single command be feasible & useful?

Let's start with an example. It is unclear to me what exactly would be
the right functionality to provide.

**Comment by anbai106** [comment]
@bthirion @GaelVaroquaux @kchawla-pi  Just to be claire what I have done: I used **image.crop_img** to crop my template MRI with a threshold 0.5 for **rtol=0.5**. Then I used the function of **image.resample_to_img** to resize my registered individual MRI to have the same dimensions with my cropped template. This works, but one disadvantage is that with **image.cro_img**, we can not really crop the neck of the skull based on the **rtol** parameter. Ideally, what I would suggest is to add the functionality to **image.resample_img** to choose a range of slices for each axis. I am not sure if I make it more clear? 

Thanks in advance

**Comment by KamalakerDadi** [comment]
@anbai106 Do you have script some where doing these steps ?

**Comment by anbai106** [comment]
@KamalakerDadi Not really, that's why I asked here
